src/aspire/source/coordinates.py

```
@@ -0,0 +1,485 @@
+        return a list of coordinates in our canonical (.box) format
+        """
+        # check if it's a STAR file list of centers
+        if os.path.splitext(coord_file)[1] == ".star":
+            return self.coords_list_from_star(coord_file)
+        # otherwise we assume text file format with one coord per line:
+        with open(coord_file, "r") as infile:
+            lines = [line.split() for line in infile.readlines()]
+        # subtract off half of particle size from center coord
+        # populate final two coordinates with the particle_size
+        return [
+            list(map(lambda x: int(x) - self.particle_size // 2, line[:2]))
+            + [self.particle_size] * 2
+            for line in lines
+        ]
+
+
+class RelionCoordinateSource(CoordinateSource):
+    """
+    Represents a data source derived from an autopick.star file within a Relion
+    project directory.
+    """
+
+    def __init__(
+        self, relion_autopick_star, particle_size, max_rows=None, dtype="double"
+    ):
+        """
+                :param files: Relion STAR file e.g. autopick.star mapping micrographs to coordinate STAR files.
+                :particle_size: Desired size of cropped particles (will override the size specified in coordinate file)
+                :param max_rows: Maximum number of particles to read. (If `None`, will
+        attempt to load all particles)
+                :param dtype: dtype with which to load images (default: double)
+        """
+
+        # if not absolute path to star file, assume relative to working dir
+        if not os.path.isabs(relion_autopick_star):
+            relion_autopick_star = os.path.join(os.getcwd(), relion_autopick_star)
+
+        # the 'coordinate_files' block of the starfile specifies
+        # paths to micrographs and coordinate files relative to the
+        # Relion project dir
+        df = StarFile(relion_autopick_star)["coordinate_files"]
```

Starfile is going to do something reasonable if relion_autopick_star has an existential crisis?

---

Would raise a KeyError for coordinate_files. I think that's alright?

---

Hrmm. let's pre-empt that crash and check if the file exists first. We can raise and error saying the file {relion_autopick_star} was not found.

---

Ok sounds good

---

Actually, `StarFile` itself will do this:

aspire/storage/starfile.py
```python
if self.filepath:                                                                                                   
            if not os.path.exists(self.filepath):                                                                           
                logger.error(f"Could not open {self.filepath}")                                                             
                raise FileNotFoundError                                                                                       
```

---

Sure, that's what I was hoping; got confused. Thanks for checking.